src/webparts/laserficheAdminConfiguration/components/ManageMappingsPage/ManageMappingsPage.tsx

```diff
@@ -327,74 +358,80 @@ export default function ManageMappingsPage(props: IManageMappingsPageProps) {
+        const restApiUrl = `${getSPListURL(
+          props.context,
+          ADMIN_CONFIGURATION_LIST
+        )}/items(${itemId})`;
+        const newMappings = [...mappings];
+        const jsonObject = JSON.stringify(newMappings);
+        const body: string = JSON.stringify({
+          Title: MANAGE_MAPPING,
+          JsonValue: jsonObject,
+        });
+        const options: ISPHttpClientOptions = {
+          headers: {
+            Accept: 'application/json;odata=nometadata',
+            'content-type': 'application/json;odata=nometadata',
+            'odata-version': '',
+            'IF-MATCH': '*',
+            'X-HTTP-Method': 'MERGE',
+          },
+          body: body,
+        };
+        await props.context.spHttpClient.post(
+          restApiUrl,
+          SPHttpClient.configurations.v1,
+          options
+        );
+        const existingSPContentType = rows.find(
+          (mapping) =>
+            mapping.SharePointContentType === rows[idx].SharePointContentType
+        );
+        if (!existingSPContentType) {
+          setValidationMessage(undefined);
+        } else {
+          setValidationMessage(validationOf);
         }
-        if (entryExists > -1) {
-          jsonValue.splice(entryExists, 1);
-          const restApiUrl = `${getSPListURL(
-            props.context,
-            ADMIN_CONFIGURATION_LIST
-          )}/items(${itemId})`;
-          const newJsonValue = [...jsonValue];
-          const jsonObject = JSON.stringify(newJsonValue);
-          const body: string = JSON.stringify({
-            Title: MANAGE_MAPPING,
-            JsonValue: jsonObject,
-          });
-          const options: ISPHttpClientOptions = {
-            headers: {
-              Accept: 'application/json;odata=nometadata',
-              'content-type': 'application/json;odata=nometadata',
-              'odata-version': '',
-              'IF-MATCH': '*',
-              'X-HTTP-Method': 'MERGE',
-            },
-            body: body,
-          };
-          props.context.spHttpClient.post(
-            restApiUrl,
-            SPHttpClient.configurations.v1,
-            options
-          );
-          for (let q = jsonValue.length + 1; q < rows.length; q++) {
-            if (
-              rows[idx].SharePointContentType == rows[q].SharePointContentType
-            ) {
-              setValidationMessage(undefined);
-              break;
-            } else {
-              setValidationMessage(validationOf);
-            }
-          }
+      } else {
+        if (mappings.length + 1 == rows.length) {
+          setValidationMessage(undefined);
         } else {
-          if (jsonValue.length + 1 == rows.length) {
-            setValidationMessage(undefined);
+          const selectSPContentType = mappings.find(
+            (mapping) => mapping.SharePointContentType === 'Select'
+          );
+          if (selectSPContentType) {
+            setValidationMessage(sharepointValidationMapping);
```

Comment:
can there exist a mapping with SharePointContentType === 'Select' AND another mapping with LaserficheContentType === 'Select'? If so, what validation message should we set?
The for loop could yield a different output than the two if statements. But it might be okay if there is only one mapping with either SharePointContentType  or LaserficheContentType  === 'Select'